src/pages/HookFormPage/HookFormPage.tsx

```diff
@@ -0,0 +1,144 @@
+import { useForm } from 'react-hook-form';
+import { useAppDispatch, useAppSelector } from '../../store/reduxHooks';
+import { useNavigate } from 'react-router-dom';
+import { addForm } from '../../store/formsDataSlice';
+import { yupResolver } from '@hookform/resolvers/yup';
+import PasswordStrength from '../../components/PasswordStrength/PasswordStrength';
+import { getSchema } from '../../utils/validation';
+
+export default function HookFormPage() {
+  const countries = useAppSelector((state) => state.countries);
+  const schema = getSchema(countries);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid, isSubmitting },
+    watch,
+  } = useForm({
+    resolver: yupResolver(schema),
+    mode: 'onChange',
+  });
+
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const password = watch('password', '');
+
+  const onSubmit = handleSubmit((data) => {
+    const file = data.image;
+    const reader = new FileReader();
+    reader.onloadend = () => {
+      const base64Image = reader.result as string;
+      const finalData = { ...data, image: base64Image };
+      dispatch(addForm(finalData));
+      navigate('/');
+    };
+    if (file?.length) {
+      reader.readAsDataURL(file[0]);
+    } else {
+      const finalData = { ...data, image: '' };
+
+      dispatch(addForm(finalData));
+      navigate('/');
+    }
+  });
+
+  return (
+    <>
+      <header className="header">
```

Comment:
It would be nice if you could add a back button to navigate back to the main page.